File: language/runtime/apple/core/Sources/RuntimeHostAppleCore/Module/Intent/IntentRequests.swift
import Foundation

/// The intent request surface attached to one Apple runtime host.
@MainActor
public protocol IntentRequests: AnyObject {
  /// Return whether one outbound URL can be opened by the Apple host.
  func canOpenURL(_ url: String) -> Bool

  /// Open one outbound URL through the Apple host.
  func openURL(_ url: String) -> UInt32

  /// Open one outbound file path through the Apple host.
  func openPath(_ path: String) -> UInt32

  /// Share one outbound text payload through the Apple host.
  func shareText(
    _ text: String,
    contentType: String?
  ) -> UInt32

  /// Share one outbound file path list through the Apple host.
  func sharePaths(
    _ paths: [String],
    contentType: String?
  ) -> UInt32
}

File: language/runtime/android/kotlin/src/main/kotlin/dev/destack/runtime/android/module/intent/IntentRequests.kt
package dev.destack.runtime.android.module.intent

/**
 * The intent request surface attached to one Android runtime host.
 */
public interface IntentRequests {
    /**
     * Return whether one outbound URL can be opened by the Android host.
     */
    public fun canOpenUrl(
        url: String,
    ): Boolean

    /**
     * Open one outbound URL through the Android host.
     */
    public fun openUrl(
        url: String,
    ): Int

    /**
     * Open one outbound file path through the Android host.
     */
    public fun openPath(
        path: String,
    ): Int

    /**
     * Share one outbound text payload through the Android host.
     */
    public fun shareText(
        text: String,
        contentType: String? = null,
    ): Int

    /**
     * Share one outbound file path list through the Android host.
     */
    public fun sharePaths(
        paths: List<String>,
        contentType: String? = null,
    ): Int
}
